youtube_dl/extractor/bandcamp.py

```diff
@@ -51,11 +75,25 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, title)
         thumbnail = self._html_search_meta('og:image', webpage, default=None)
         m_download = re.search(r'freeDownloadPage: "(.*?)"', webpage)
+        m_trackinfo = re.search(r'trackinfo: (.+),\s*?\n', webpage)
+        json_code = m_trackinfo.group(1) if m_trackinfo else None
+        data = json.loads(json_code)[0] if json_code else None
+
+        match = re.search(r'album_title\s*:\s*"([^"]+)"', webpage)
+        album_title = match.group(1) if match else None
+
+        match = re.search(r'artist\s*:\s*"([^"]+)"', webpage)
+        artist = match.group(1) if match else None
+
+        match = re.search(r'album_release_date\s*:\s*"([^"]+)"', webpage)
+        release_date = unified_strdate(match.group(1)) if match else None
+        release_year = int(release_date[0:4]) if release_date else None
```

Comment:
Breaks extraction if `release_date[0:4]` is not `int`.

Reply:
It's supposed to hold YYYYMMDD as per the documentation of `unified_strdate`.

Reply:
So what? You extract `release_date` from a `webpage`'s `album_release_date`. You have **absolutely no guarantee** `album_release_date` will match `YYYYMMDD`.

Reply:
It's formatted as RFC 2822.